src/transport/HttpClient.ts

```diff
@@ -8,6 +8,12 @@ interface HttpClientBase {
     getRecentQuizzes(): Promise<QuizSummary[]>;
 
     getQuizDetails(workspaceId: string, quizId: number): Promise<QuizDetails>;
+
+    getWorkspaces(): Promise<Workspace[]>;
+
+    verifyNoteIdIdentity(workspaceId: string, noteId: string): Promise<boolean>;
+
+    verifyQuizIdIdentity(quizId: string): Promise<boolean>;
```

Comment:
i hope it won't be problematic to implement in the future

Reply:
why could it be problematic? just a db query for existance by id. should be pretty easy